symengine/complex_mpc.cpp

```diff
@@ -908,6 +908,38 @@ class EvaluateMPC : public Evaluate
                 MPFR_RNDN);
         return complex_mpc(std::move(t));
     }
+    virtual RCP<const Basic> floor(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<ComplexMPC>(x))
+        integer_class re, im;
+        mpfr_get_z(
+            get_mpz_t(re),
+            mpc_realref(down_cast<const ComplexMPC &>(x).as_mpc().get_mpc_t()),
+            MPFR_RNDD);
+        mpfr_get_z(
+            get_mpz_t(im),
```

Comment:
If you use `get_mpz_t` for input, you need to do `mp_demote` afterwards. See `mp_class.h` for some examples.